Add SkipClean option

jingweno opened the pull request:
This PR rebases https://github.com/gorilla/mux/pull/54 on the latest master. Most works were done by @whatupdave.

/cc @elithrar

elithrar commented on route.go:
```diff
@@ -26,6 +26,9 @@ type Route struct {
 	strictSlash bool
+	// If true, when the path pattern is "/path//to", accessing "/path//to"
+	// will not redirect
+	skipClean bool
```
We should still add `SkipClean()` as a method on `*Route` as well.

elithrar commented on mux.go:
```diff
@@ -133,6 +137,19 @@ func (r *Router) StrictSlash(value bool) *Router {
 }
 
+// SkipClean defines the path cleaning behaviour for new routes. The initial
+// value is false.
```
`SkipClean defines the path cleaning behaviour for new routes. The initial value is false. Users should be careful about which routes are not cleaned`.

elithrar commented:
Ping @jingweno in case you didn't see my line notes.

jingweno commented:
@elithrar Sorry for the late response. I addressed all comments and added a test. Could we merge this if it looks okay? I'd need the change to fix a production issue. Thanks.

jingweno commented:
This closes https://github.com/gorilla/mux/pull/54.

jingweno commented:
I removed installing `go-vet` as part of the build since it's now part of Go std distro: https://travis-ci.org/gorilla/mux/builds/127283267.

elithrar commented:
Sure, looking at it now!

jingweno commented:
Thanks @elithrar!

elithrar commented:
LGTM. Thansk @jingweno!
